frontend/src/AppInbusImport.vue

```diff
@@ -0,0 +1,327 @@
+  const headers: Headers = new Headers();
+
+  headers.set('Content-Type', 'application/json');
+  headers.set('Accept', 'application/json');
+
+  const request: Request = new Request(url, {
+    method: 'GET',
+    headers: headers
+  });
+
+  return request;
+}
+
+function svcc2num(svcc: string): number {
+  const [dept_code, version]: string[] = svcc.split('/');
+  const [dept, code]: string[] = dept_code.split('-');
+
+  const svcc_str = dept + code + version;
+  const svcc_num = Number(svcc_str);
+
+  return svcc_num;
+}
+
+async function loadInbusAndKelvinSubjects() {
+  const res1 = await fetch('/api/inbus/subject_versions', {});
+  const res2 = await fetch('/api/subjects/all', {});
+
+  subjects_inbus = await res1.json();
+  const subjects_kelvin_resp = await res2.json();
+
+  subjects_kelvin = subjects_kelvin_resp.subjects;
+  subjects_kelvin.sort((a, b) => {
+    const name_a = a.name.toUpperCase();
+    const name_b = b.name.toUpperCase();
+    if (name_a < name_b) {
+      return -1;
+    }
+    if (name_a > name_b) {
+      return 1;
+    }
+    return 0;
+  });
+
+  const subject_kelvin_abbrs = subjects_kelvin.map((s) => s.abbr);
+
+  subjects_inbus_filtered = subjects_inbus.filter((subject_inbus) =>
+    subject_kelvin_abbrs.includes(subject_inbus.subject.abbrev)
+  );
+  subjects_inbus_filtered = subjects_inbus_filtered.sort(
+    (a, b) => svcc2num(a.subjectVersionCompleteCode) - svcc2num(b.subjectVersionCompleteCode)
+  );
+
+  inbus_and_kelvin_subjects_loading.value = false;
+}
+
+function parseSemesters(semesters_data: Semester[]) {
+  semesters.value = semesters_data.map((sm) => ({
+    pk: sm.pk,
+    year: sm.year,
+    winter: sm.winter,
+    display: String(sm.year) + (sm.winter ? 'W' : 'S')
+  }));
+  semesters_loading.value = false;
+}
+
+async function loadSemesters() {
+  const request = assembleRequest('/api/semesters');
+  const res = await fetch(request, {});
+
+  const semesters_data = await res.json();
+  parseSemesters(semesters_data['semesters']);
+}
+
+async function loadScheduleForSubjectVersionId(subject_inbus: InbusSubjectVersion) {
+  const request = assembleRequest(
+    '/api/inbus/schedule/subject/version/' + subject_inbus.subjectVersionId
+  );
+  let res = await fetch(request, {});
+  subject_inbus_schedule.value = await res.json();
+  inbus_schedule_loaded.value = true;
+}
+
+async function importActivities() {
+  busy.value = true;
+  const req = {
+    semester_id: semester.value,
+    subject: subject_kelvin_selected.value,
+    activities: classes_to_import.value
+  };
+
+  //console.log(req);
+
+  const res = await fetch('/api/import/activities', {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+      'X-CSRFToken': csrfToken()
+    },
+    body: JSON.stringify(req)
+  });
+
+  result.value = await res.json();
+  busy.value = false;
+}
+
+onMounted(() => {
+  watch(subject_inbus_selected, (newValue, oldValue) => {
```

Comment:
I'd put the `watch`es onto the top level, to make them more visible. They don't need to be created inside `onMounted`.

Reply:
One moved to the top level, one removed.